rs/replica/setup_ic_network/src/lib.rs

```diff
@@ -85,6 +85,105 @@ struct ArtifactPools {
     canister_http_pool: Arc<RwLock<CanisterHttpPoolImpl>>,
 }
 
+impl ArtifactPools {
+    fn new(
+        log: &ReplicaLogger,
+        metrics_registry: &MetricsRegistry,
+        node_id: NodeId,
+        config: ArtifactPoolConfig,
+        catch_up_package: &CatchUpPackage,
+    ) -> Self {
+        let ingress_pool = Arc::new(RwLock::new(IngressPoolImpl::new(
+            node_id,
+            config.clone(),
+            metrics_registry.clone(),
+            log.clone(),
+        )));
+
+        let mut idkg_pool = IDkgPoolImpl::new(
+            config.clone(),
+            log.clone(),
+            metrics_registry.clone(),
+            Box::new(idkg::IDkgStatsImpl::new(metrics_registry.clone())),
+        );
+        idkg_pool.add_initial_dealings(catch_up_package);
+        let idkg_pool = Arc::new(RwLock::new(idkg_pool));
+
+        let certification_pool = Arc::new(RwLock::new(CertificationPoolImpl::new(
+            node_id,
+            config,
+            log.clone(),
+            metrics_registry.clone(),
+        )));
+        let dkg_pool = Arc::new(RwLock::new(DkgPoolImpl::new(
+            metrics_registry.clone(),
+            log.clone(),
+        )));
+        let canister_http_pool = Arc::new(RwLock::new(CanisterHttpPoolImpl::new(
+            metrics_registry.clone(),
+            log.clone(),
+        )));
+        Self {
+            ingress_pool,
+            certification_pool,
+            dkg_pool,
+            idkg_pool,
+            canister_http_pool,
+        }
+    }
+}
+
+struct Bouncers {
+    ingress: Arc<IngressBouncer>,
+    consensus: Arc<ConsensusBouncer>,
+    certifier: Arc<CertifierBouncer>,
+    dkg: Arc<DkgBouncer>,
+    idkg: Arc<idkg::IDkgBouncer>,
+    https_outcalls: Arc<CanisterHttpGossipImpl>,
```

Comment:
I think calling it `canister_http` would be more consistent with the naming in this file.
```suggestion
    canister_http: Arc<CanisterHttpGossipImpl>,
```

Reply:
Let me keep it like this, and I will rename more in a follow-up PR so things are more consistent.